scipy/stats/_stats_py.py

```diff
@@ -805,11 +806,15 @@ def tmin(a, lowerlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
     a, mask = _put_val_to_limits(a, (lowerlimit, None), (inclusive, None),
                                  val=max_, xp=xp)
 
-    min_ = xp.min(a, axis=axis)
-    valid = ~xp.all(mask, axis=axis)  # At least one element above lowerlimit
-    # Output dtype is data-dependent
-    # Possible loss of precision for int types
-    res = min_ if xp.all(valid) else xp.where(valid, min_, xp.nan)
+    res = xp.min(a, axis=axis)
+    invalid = xp.all(mask, axis=axis)  # All elements are below lowerlimit
+
+    if is_lazy_array(invalid) or xp.any(invalid):
+        # For eager backends, output dtype is data-dependent
+        # Possible loss of precision for int types
+        res, = xp_broadcast_promote(res, force_floating=True, xp=xp)
```

Comment:
Looks good, but I'd like to get gh-22695 merged first to avoid the conflict (`xp_broadcast_promote` will no longer exist).